PSReadLine/ReadLine.cs

```diff
@@ -196,12 +200,17 @@ internal static PSKeyInfo ReadKey()
                     //   - a key is pressed
                     //   - the console is exiting
                     //   - 300ms - to process events if we're idle
-                    //   - processing of events is requested externally
                     //   - ReadLine cancellation is requested externally
                     handleId = WaitHandle.WaitAny(_singleton._requestKeyWaitHandles, 300);
                     if (handleId != WaitHandle.WaitTimeout && handleId != EventProcessingRequested)
                         break;
 
+                    if (_handleIdleOverride is not null)
+                    {
+                        _handleIdleOverride(_singleton._cancelReadCancellationToken);
```

Comment:
If you use the cancellation token to break out from `WaitHandle.WaitAny`, then `_cancelReadCancellationToken` would have been in the "cancellation requested" state, then what's the point passing it in?